### PR TITLE
added features to artist post route

### PR DESCRIPTION
**Pull request by PraiseTy**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1 +1,5 @@
-MONGO_URI =
\ No newline at end of file
+MONGO_URI =
+
+JWT_SECRET =
+
+JWT_LIFETIME = 12h
```
Comment (greatertomi) [inline .env.example]:
```suggestion
MONGO_URI =
JWT_SECRET =
JWT_LIFETIME = 12h
```

File: controllers/artists.js [hunk 2]
```diff
@@ -1,23 +1,75 @@
 const Artist = require('../models/Artist');
+const HTTP_STATUS = require('../constant');
+const logger = require('../logging/logger');
 
 const createNewArtist = async (req, res) => {
-  res.json('Create new artists');
+  try {
+    const artist = await Artist.create({ ...req.body });
```
Comment (greatertomi) [inline controllers/artists.js]: This exposes your model to receiving values you don't expect and can't control. Destructure what you need and pass it here.

```
const {email,...} = req.body
Artist.create({email,...})
```

File: controllers/artists.js
Comment (greatertomi) [inline controllers/artists.js:1]: It is a good naming pattern to add the role of a file to its name. Rename all your files to follow this pattern
- artists.controller.js
- artists.route.js
- Artist.model.js

File: controllers/artists.js [hunk 3]
```diff
@@ -1,23 +1,75 @@
 const Artist = require('../models/Artist');
+const HTTP_STATUS = require('../constant');
+const logger = require('../logging/logger');
```
Comment (greatertomi) [inline controllers/artists.js]: Remove

**Review by greatertomi — CHANGES_REQUESTED** [review]
Generally. Good stuff.

Please fix the eslint issues in your code. For example this
> ESLint: Expected to return a value at the end of async arrow function.(consistent-return)

File: controllers/artists.js [hunk 4]
```diff
@@ -1,23 +1,75 @@
...
 
 const createNewArtist = async (req, res) => {
-  res.json('Create new artists');
+  try {
+    const artist = await Artist.create({ ...req.body });
+    res.status(HTTP_STATUS.CREATED).json({
+      message: 'Artist created successfully',
+      data: { id: artist._id, name: artist.name }
```
Comment (greatertomi) [inline controllers/artists.js]: This line throws an unnecessary error. You can switch off `no-underscore-dangle` within eslint

File: controllers/artists.js (hunk 2 above)
Comment (PraiseTy) [reply controllers/artists.js:7]: done on a332db4

File: .env.example (hunk 1 above)
Comment (PraiseTy) [reply .env.example]: done on a332db4

File: controllers/artists.js (hunk 2 above)
Comment (PraiseTy) [reply controllers/artists.js]: done on baf056f

File: controllers/artists.js
Comment (PraiseTy) [reply controllers/artists.js:1]: done on e3a5cc0

File: controllers/artists.js (hunk 3 above)
Comment (PraiseTy) [reply controllers/artists.js]: done on 323ea2a

File: controllers/artists.js (hunk 4 above)
Comment (PraiseTy) [reply controllers/artists.js]: done on e66a553

**Review by greatertomi — APPROVED** [review]

**Pull request merged**